test/gurka/test_turn_lanes.cc

```diff
@@ -43,3 +59,240 @@ TEST(Standalone, TurnLanes) {
   EXPECT_EQ(prev_edge->turn_lanes_size(), 3);
   EXPECT_EQ(prev_edge->TurnLanesToString(), "[ left | through | through;right ACTIVE ]");
 }
+
+// Split lane example - 5-way intersection
+// (40.856684, -73.869336)
+TEST(Standalone, TurnLanesSplitLane) {
+
+  const std::string ascii_map = R"(
+        C
+        |
+        |  D--E
+    A---B<
+        |  F--G
+        |
+        H
+  )";
+
+  const gurka::ways ways = {{"AB",
+                             {{"highway", "trunk"},
+                              {"name", "Bronx and Pelham Parkway"},
+                              {"oneway", "yes"},
+                              {"lanes", "3"},
+                              {"turn:lanes", "left|through;left|through;right"}}},
+                            {"CBH", {{"highway", "primary"}, {"name", "Boston Road"}}},
+                            {"BDE",
+                             {{"highway", "trunk"},
+                              {"name", "Bronx and Pelham Parkway"},
+                              {"oneway", "yes"},
+                              {"lanes", "3"},
+                              {"turn:lanes", "through|through|through"}}},
+                            {"BFG",
+                             {{"highway", "tertiary"},
+                              {"name", "Pelham Parkway South"},
+                              {"oneway", "yes"},
+                              {"lanes", "1"},
+                              {"turn:lanes", "through"}}}};
+
+  const auto layout = gurka::detail::map_to_coordinates(ascii_map, 10);
+  auto map = gurka::buildtiles(layout, ways, {}, {}, "test/data/gurka_turn_lanes_2");
+  valhalla::Api result;
+
+  // A -> C - takes the leftmost left lane
+  result = gurka::route(map, "A", "C", "auto");
+  validate_turn_lanes(result, {
+                                  {3, "[ left ACTIVE | left;through | through;right ]"},
+                                  {0, ""},
+                              });
+
+  // A -> E - takes the leftmost through lane
+  result = gurka::route(map, "A", "E", "auto");
+  validate_turn_lanes(result, {
+                                  {3, "[ left | left;through ACTIVE | through;right ]"},
+                                  {0, ""}, // TODO lanes are tossed when all are through
+                              });
+
+  // A -> G - takes the rightmost through lane
+  result = gurka::route(map, "A", "G", "auto");
+  gurka::assert::osrm::expect_steps(result, {"Bronx and Pelham Parkway", "Pelham Parkway South"});
```

Comment:
Why does only this route have extra validations for steps & maneuvers but others don't?

Reply:
This test map had street names to help distinguish the two parallel roads, so I used those to verify that the correct parallel road was taken.